src/zkbob/manager/MPCGuard.sol

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "../../../src/zkbob/ZkBobPool.sol";
+import "../../utils/Ownable.sol";
+import "../utils/CustomABIDecoder.sol";
+
+import "../../interfaces/IZkBobPool.sol";
+
+contract MPCGuard is Ownable, CustomABIDecoder {
+    address[] private guards;
+
+    address operator;
+
+    address public immutable pool;
+
+    uint256 constant SIGNATURE_SIZE = 64;
+
+    constructor(address _operator, address _pool) Ownable() {
+        pool = _pool;
+        _setOperator(_operator);
+    }
+
+    /**
+     * @dev Throws if called by any account other than the current relayer operator.
+     */
+    modifier onlyOperator() {
+        require(operator == _msgSender(), "ZkBobPool: not an operator");
+        _;
+    }
+
+    function _setOperator(address _operator) internal {
+        operator = _operator;
+    }
+
+    function setOperator(address _operator) external onlyOwner {
+        _setOperator(_operator);
+    }
+
+    function setGuards(address[] calldata _guards) external onlyOwner {
```

Comment:
Let's verify that the guards list is sorted, (e.g. `0xA..A`, `0xB..B`, `0xC..C`). This will make it easier to migrate to other quorum scheme (e.g. N-of-M), if necessary.

Reply:
fixed in https://github.com/zkBob/zkbob-contracts/pull/99/commits/13237ca8b15e0912f4047bfbea83d47f71dd417c